modules/parent-join/src/main/java/org/elasticsearch/join/query/HasChildQueryBuilder.java

```diff
@@ -136,6 +140,9 @@ public HasChildQueryBuilder minMaxChildren(int minChildren, int maxChildren) {
         if (minChildren < 0) {
             throw new IllegalArgumentException("[" + NAME + "] requires non-negative 'min_children' field");
         }
+        if (minChildren == 0) {
```

Comment:
Since maxChildren == 0 is also disallowed in 8.0, that should get a warning too.

Reply:
@cbuescher do you really think we need this warning? if maxChildren is 0 then either minChildren will be 0 so we will get a warning for it, or maxChildren < minChildren will throw an error. That's what I thought, what do you think?

Reply:
I updated DEFAULT_MIN_CHILDREN to be 1 at the place of 0. So the case where min_children is not provided but max_children is set to 0, this throws an error. Do you think I should keep DEFAULT_MIN_CHILDREN 0, this will cause CI to fail because they'll all be throwing warnings?

Reply:
So basically we have 2 solutions here:
1- Put DEFAULT_MIN_CHILDREN back to 0, and fix all CI tests not to cause the warnings, and add the warning for maxChildren == 0

2- Keep DEFAULT_MIN_CHILDREN 1, at the risk of breaking queries for people who might be providing a query like this 1:
```
{
  "query": {
    "has_child": {
      "type":"answer",
      "max_children": 0, 
      "query": {
        "match_all": {}
      }
    }
  }
}'
```

Reply:
one last detail to consider, is that if we go with solution 1, the deprecation warning will be displayed when the query doesn't have min_children, which will cause the user to think that min_children is mandatory, which is not the case. (that's why I updated DEFAULT_MIN_CHILDREN). sorry for spamming :) I am brainstorming here

Reply:
Thanks for the explanation, I think you are right in updating the  DEFAULT_MIN_CHILDREN and this will give sufficient warning if we only check the min value, otherwise the later maxChildren < minChildren condition will trigger an error. The case where "minChildren" is unset and "maxChildren" set to 0 is highly unlikely and can be neglected I think.